CleanArchitecture/Program.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Bogus;
+using CleanArchitecture;
+using Microsoft.Azure.CosmosRepository;
+using Microsoft.Azure.CosmosRepository.CleanArchitecture;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+ConfigurationBuilder configuration = new();
+
+ServiceProvider provider = new ServiceCollection().AddCosmosRepository(options =>
+    {
+        options.CosmosConnectionString = Environment.GetEnvironmentVariable("CosmosConnectionString");
+        options.DatabaseId = "paging-db";
+        options.ContainerPerItemType = true;
+        options.OptimizeBandwidth = false;
+    })
+    .AddSingleton<IConfiguration>(configuration.Build())
+    .WithEtagMappedRepositories()
+    .AddSingleton<IItemMapper<PersonItem, PersonEntity>, PersonItemMapper>()
+    .AddSingleton<IMappedItemMapper<PersonEntity, PersonItem>, PersonEntityMapper>()
+    .BuildServiceProvider();
+
+IRepository<PersonItem> globalRepo = provider.GetRequiredService<IRepository<PersonItem>>();
+
+await SeedAsync();
```

Comment:
Would be good to see how this would work in a wrapped repository, say you defined an `IPersonRepository` with a method like `ValueTask<Person> GetAsync(string id)`

Would the caller still benefit from this storage of the ETag?

Reply:
I have updated the CleanArch sample to show how to do this without corrupting the layers with infrastructure references.